### DISCUSSION TITLE
I'd like to add a custom gitignore template

### DISCUSSION THREAD

**Discussion by k3davis** [community/community#62107] [Discussions]
### Select Topic Area

Product Feedback

### Body

Our team uses a default gitignore file for most of its repositories, that is different from the delivered ones because the devs use different IDEs, languages and/or operating systems at times. We would love to have the ability to have a custom default gitignore at the organization level, not necessarily to replace existing options, but as an additional choice. Can you consider adding this feature, or is there an alternative I've overlooked? Thank you.

**Comment by CodeMaster7000** [comment]
Hi @k3davis 

This feature has been requested before on Gitextensions, and it turns out that many of us had been overlooking it the whole time! Simply click here: https://github.com/gitextensions/gitextensions/issues/7487 and scroll down to the bottom where you can find out how to use the feature.

Have a great day and happy coding!

**Reply by srce-code** [reply]
To use the mentioned solution, we will need to install GitHub Desktop, then GitHub Extensions, and then create/edit the mentioned file. Some of us don't use those programs. Additionally, this won't affect the gitignore template set when creating a new repo on the GitHub website (in the dropdown under "Add .gitignore"). I'd specifically like to see this customization capability in GitHub.

Basically, I also want this feature! Ideally we'd be able to customize the list of gitignore templates, with ability to exclude defaults as well as include new custom/private ones. I don't care about publishing mine for others, just want to be able to create a custom set & use those options. Similar to how we can create our own template repos and use those.

(Sorry I know this is old but I don't think the other solutions really address this.)

**Comment by dimasinr** [comment]
u can use .gitignore_global: Git supports a global .gitignore file that applies to all Git repositories on your local machine. While this won't enforce organization-wide standards, you can encourage developers to set up a global .gitignore file with common patterns.
